packages/chips/ChipSet.js

```diff
@@ -41,6 +54,20 @@ export default class ChipSet extends Component {
     };
   }
 
+  handleSelect = (chipId) => {
+    if (this.props.filter || this.props.choice) {
+      this.foundation_.toggleSelect(chipId);
+    }
+    this.props.handleSelect(this.foundation_.getSelectedChipIds());
+  }
+
+  handleRemove = (chipId) => {
```

Comment:
Bypassing `foundation.handleChipRemoval` so we don't need to pass the id in `event.details`